### PR TITLE
add a utility which when called upon changes nickname

### PR DESCRIPTION
**Pull request by RitikJaiswal75**
Created a utility function named `updateNickName` 

The function calls up on the `/guilds/DISCORD_GUILD_ID/members/discordId`
The function takes in 3 parameters:
- discord_Id: String
- nickname: String (desired nickname)
- env: Object which contains DISCORD_TOKEN, GUILD_ID

It is a patch call made to discord's API with a body of `{nick: nickname}`

### CODE REVIEW

File: src/utils/updateNickname.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    headers: {
+      "Content-Type": "application/json",
+      Authorization: `Bot ${env.DISCORD_TOKEN}`,
+    },
+    body: JSON.stringify(data),
+  });
+  if (nameChangeResponse.ok) {
+    return await nameChangeResponse.json();
+  }
+  return "Failed";
```
Comment (heyrandhir) [inline src/utils/updateNickname.ts:22]: we can be a bit more specific here. Just `failed` doesn't help much in knowing why this call failed.

**Review by heyrandhir — APPROVED** [review]

File: src/utils/updateNickname.ts (hunk 1 above)
Comment (RitikJaiswal75) [reply src/utils/updateNickname.ts]: updated

File: src/utils/updateNickname.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bot ${env.DISCORD_TOKEN}`,
+      },
+      body: JSON.stringify(data),
+    });
+    if (nameChangeResponse.ok) {
+      return await nameChangeResponse.json();
+    }
+    return "Some internal Server error occurred";
```
Comment (samyakshah3008) [inline src/utils/updateNickname.ts:23]: Can we write this in constant instead of hard coding the error message?

**Review by samyakshah3008 — DISMISSED** [review]
Great work! 
Added a comment, please check. 
Rest is all good!

**Review by heyrandhir — DISMISSED** [review]

File: src/utils/updateNickname.ts (hunk 2 above)
Comment (yesyash) [inline src/utils/updateNickname.ts:23]: If the response is not okay and we return with `Some internal....` the console log inside the catch block will not be executed. 

Is this flow right and in the case of the response not being okay do we not want to log that action?

Comment (RitikJaiswal75) [reply src/utils/updateNickname.ts:23]: oh, That sounds right. I made the changes and updated the PR

Comment (RitikJaiswal75) [reply src/utils/updateNickname.ts:23]: Done and updated with the PR

**Review by heyrandhir — DISMISSED** [review]

File: src/utils/updateNickname.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      method: "PATCH",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bot ${env.DISCORD_TOKEN}`,
+      },
+      body: JSON.stringify(data),
+    });
+    if (nameChangeResponse.ok) {
+      return await nameChangeResponse.json();
+    }
```
Comment (yesyash) [inline src/utils/updateNickname.ts:23]: What happens if the response is not ok? 
Are we not handling that case?

Comment (RitikJaiswal75) [reply src/utils/updateNickname.ts:23]: it should throw an error in that case which would be caught in catch block, am I wrong here?

Comment (samyakshah3008) [reply src/utils/updateNickname.ts:23]: Same question as @yesyash

**Review by samyakshah3008 — APPROVED** [review]
Looks great to merge! 
Added one question, rest is perfect

File: src/utils/updateNickname.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import { INTERNAL_SERVER_ERROR } from "../constants/responses";
+import { DISCORD_BASE_URL } from "../constants/urls";
+import { env } from "../typeDefinitions/default.types";
+
+export async function updateNickName(
+  discordId: string,
+  nickname: string,
+  env: env
+) {
+  const changeNickNameURL = `${DISCORD_BASE_URL}/guilds/${env.DISCORD_GUILD_ID}/members/${discordId}`;
```
Comment (Maheima) [inline src/utils/updateNickname.ts:10]: can this URL be moved to a constants file and its casing be changed to uppercase snake cased letters?

**Review by Maheima — CHANGES_REQUESTED** [review]

File: src/utils/updateNickname.ts (hunk 4 above)
Comment (RitikJaiswal75) [reply src/utils/updateNickname.ts:10]: This URL is formed from user input how can it be a constant? it's different for every user. As of what I know Constants are not re-assignable after initialization. Could you point me somewhere where it says something different?
I have used a part of the URL from constants i.e `DISCORD_BASE_URL`
If it is not in the constants file do we need the uppercase snake case?

File: src/utils/updateNickname.ts (hunk 3 above)
Comment (RitikJaiswal75) [reply src/utils/updateNickname.ts]: Hey @yesyash I have updated the PR.

**Review by shubhamsigdar1 — APPROVED** [review]

**Review by rohan09-raj — APPROVED** [review]

**Pull request merged**